### PR TITLE
[core] Rationalize style property classes

### PR DESCRIPTION
**Pull request by jfirebaugh**
This is the first in a series of PRs that builds towards a full-fledged [programmatic style API](https://github.com/mapbox/mapbox-gl-native/issues/837). This is a big chunk of work, but I'm trying to break it up step by step and not create a long-lived branch with a lot of changes.

Overall, my first major goal is to eliminate `ClassProperties` -- this is essentially an untyped map of every possible style property. Instead, we should work with classes like:

```cpp
class FillPaintProperties {
public:
    bool antialias = true;
    float opacity = 1.0f;
    Color fill_color = {{ 0, 0, 0, 1 }};
    Color stroke_color = {{ 0, 0, 0, -1 }};
    std::array<float, 2> translate = {{ 0, 0 }};
    TranslateAnchorType translateAnchor = TranslateAnchorType::Map;
    Faded<std::string> image;

    inline bool isVisible() const {
        return opacity > 0 && (fill_color[3] > 0 || stroke_color[3] > 0);
    }
};
```

These classes already exist, they are just inconsistently named. This PR makes the naming both internally consistent, and consistent with terminology from the style specification: for each layer type, there is a `____PaintProperties` class and a `____LayoutProperties` class. For the time being, all of these classes live in `style_properties.hpp`.

In code that I touch, I'm also fixing style issues like `&` placement and variable naming conventions. This will be the case with the whole series of PRs.

:eyes: @kkaefer @brunoabinader

### CODE REVIEW

File: include/mbgl/style/style_properties.hpp [hunk 1]
```diff
@@ -105,26 +160,31 @@ struct RasterProperties {
...
-    RasterProperties,
-    BackgroundProperties,
+    FillPaintProperties,
+    LinePaintProperties,
+    CirclePaintProperties,
+    SymbolPaintProperties,
+    RasterPaintProperties,
+    BackgroundPaintProperties,
     std::false_type
 > StyleProperties;
```
Comment (kkaefer) [inline include/mbgl/style/style_properties.hpp]: Can we rename this to `StylePaintProperties`?

**Comment by kkaefer** [comment]
Looks good!

**Comment by brunoabinader** [comment]
It is great that now layout properties are now exported to the [public API](https://github.com/mapbox/mapbox-gl-native/pull/2705/files#diff-22f9c5df9ed2491e62718e2d7b68652b). :+1: !

File: include/mbgl/style/style_properties.hpp (hunk 1 above)
Comment (jfirebaugh) [inline include/mbgl/style/style_properties.hpp]: It's going away in the next commit, so I held off.

**Pull request merged**